svm/tests/integration_test.rs

```diff
@@ -1,241 +1,499 @@
+// one transaction in a batch plus check results for svm and asserts for tests
+#[derive(Clone, Debug)]
+pub struct TransactionBatchItem {
+    pub transaction: Transaction,
+    pub check_result: TransactionCheckResult,
+    pub asserts: TransactionBatchItemAsserts,
 }
 
-#[test]
-fn svm_integration() {
+impl Default for TransactionBatchItem {
+    fn default() -> Self {
+        Self {
+            transaction: Transaction::default(),
+            check_result: Ok(CheckedTransactionDetails {
+                nonce: None,
+                lamports_per_signature: 5000,
+            }),
+            asserts: TransactionBatchItemAsserts::succeeded(),
+        }
+    }
+}
+
+// asserts for a given transaction in a batch
+// we can automatically check whether it executed, whether it succeeded
+// log items we expect to see (exect match only), and rodata
+// rodata is a doubly-nested Option. None means no check is performed, Some(None) means we assert no rodata
+#[derive(Clone, Debug)]
+pub struct TransactionBatchItemAsserts {
+    pub executed: bool,
+    pub succeeded: bool,
+    pub logs: Vec<String>,
+    pub return_data: Option<Option<TransactionReturnData>>,
+}
+
+impl TransactionBatchItemAsserts {
+    pub fn succeeded() -> Self {
+        Self {
+            executed: true,
+            succeeded: true,
+            logs: vec![],
+            return_data: None,
+        }
+    }
+
+    pub fn failed() -> Self {
+        Self {
+            executed: true,
+            succeeded: false,
+            logs: vec![],
+            return_data: None,
+        }
+    }
+
+    pub fn not_executed() -> Self {
+        Self {
+            executed: false,
+            succeeded: false,
+            logs: vec![],
+            return_data: None,
+        }
+    }
+}
+
+fn program_medley() -> TestData {
+    let mut test_data = TestData::default();
+
+    // 0: A transaction that works without any account
+    {
+        let program_name = "hello-solana".to_string();
+        let program_id = program_address(&program_name);
+        test_data
+            .initial_programs
+            .push((program_name, DEPLOYMENT_SLOT));
+
+        let fee_payer_keypair = Keypair::new();
+        let fee_payer = fee_payer_keypair.pubkey();
+
+        let mut fee_payer_data = AccountSharedData::default();
+        fee_payer_data.set_lamports(LAMPORTS_PER_SOL);
+        test_data.add_initial_account(fee_payer, &fee_payer_data);
+
+        let instruction = Instruction::new_with_bytes(program_id, &[], vec![]);
+        test_data.push_transaction(Transaction::new_signed_with_payer(
+            &[instruction],
+            Some(&fee_payer),
+            &[&fee_payer_keypair],
+            Hash::default(),
+        ));
+
+        test_data.transaction_batch[0]
+            .asserts
+            .logs
+            .push("Program log: Hello, Solana!".to_string());
+
+        test_data.sub_lamports(&fee_payer, LAMPORTS_PER_SIGNATURE);
```

Comment:
I think the name of these functions could be clearer and say that we are modifying the account state we expect and the end of the execution. From just laying eyes in this code snippet, it looks like you are changing the account before execution.